File: workers/business-worker/src/worker.ts
import { Worker } from "@temporalio/worker";
import * as activities from "./activities";

// TS 业务 Worker（ADR-004/005）：执行确定性 Activity，订阅 logix-business 任务队列。
async function run(): Promise<void> {
  const worker = await Worker.create({
    workflowsPath: require.resolve("./workflows"),
    activities,
    taskQueue: "logix-business",
  });
  console.log("business-worker listening on task queue: logix-business");
  await worker.run();
}

run().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: workers/ai-worker/worker.py
# Python AI Worker（ADR-005）：执行 AI Activity，调用 ai-service 能力。
# 骨架阶段：echo 工作流 + echo AI Activity；真实 AI 能力在 P6/P7 由业务工作流跨队列调用。
import asyncio

from temporalio.client import Client
from temporalio.worker import Worker

import activities
import workflows


async def main() -> None:
    client = await Client.connect("localhost:7233")
    worker = Worker(
        client,
        task_queue="logix-ai",
        workflows=[workflows.EchoAiWorkflow],
        activities=[activities.echo_ai_activity],
    )
    print("ai-worker listening on task queue: logix-ai")
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())
